week10/crud/src/pages/Register.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { postRegister, postLogin } from '../api/user';
+import styled from 'styled-components';
+import client from '../api';
+
+function RegisterPage() {
+    const navigate = useNavigate();
+
+    // 회원가입 상태 관리
+    const [signup, setSignup] = useState({ id: "", pw: "", nickname: "" });
+
+    const onChangeSignup = (e) => {
+        const { name, value } = e.target;
+        setSignup({ ...signup, [name]: value });
+    };
+
+    const handleSignup = async (e) => {
+        e.preventDefault();
+        const { id, pw, nickname } = signup;
+        const res = await postRegister(id, pw, nickname);
+        if (res.status === 201) {
+            alert("회원가입 성공! 로그인을 진행해주세요.");
+            setSignup({ id: "", pw: "", nickname: "" });
+        } else if (res.response.status === 400) {
+            alert(res.response.data.message);
+        }
+    };
+
+    // 로그인 상태 관리
+    const [login, setLogin] = useState({ id: "", pw: "" });
+
+    const onChangeLogin = (e) => {
+        const { name, value } = e.target;
+        setLogin({ ...login, [name]: value });
+    };
+
+    const handleLogin = async (e) => {
+        e.preventDefault();
+        const { id, pw } = login;
+        const res = await postLogin(id, pw);
+        if (res.status === 201) {
+            alert(`로그인 성공! ${res.data.username}님 환영합니다.`);
+            localStorage.setItem("efubtoken", res.data.token);
+            client.defaults.headers.common["Authorization"] = res.data.token; 
+            navigate("/");
+            window.location.reload();
+        } else if (res.response.status === 400) {
+            alert("아이디와 비밀번호가 맞는지 다시 확인해주세요!");
+        }
+    };

```

Comment:
- [x] 2-2. try ~ catch 문으로 예외 처리 강화
```suggestion
const handleLogin = async (e) => {
    e.preventDefault();
    const { id, pw } = login;
    try {
      const res = await postLogin(id, pw);
      if (res.status === 201) {
        alert(`로그인 성공! ${res.data.username}님 환영합니다.`);
        localStorage.setItem("efubtoken", res.data.token);
        client.defaults.headers.common["Authorization"] = res.data.token;
        navigate("/");
        window.location.reload();
      } else if (res.response.status === 400) {
        alert("아이디와 비밀번호가 맞는지 다시 확인해주세요!");
      }
    } catch (error) {
      console.error("로그인 오류", error);
    }
  };
```
handleLogin에 적용한 예시입니다.